homeassistant/components/ezviz/config_flow.py

```diff
@@ -71,89 +90,43 @@ class EzvizConfigFlow(ConfigFlow, domain=DOMAIN):
-
-        except HTTPError as err:
-            raise InvalidHost from err
-
-        except PyEzvizError as err:
-            raise PyEzvizError from err
-
-        auth_data = {
-            CONF_USERNAME: data[CONF_USERNAME],
-            CONF_PASSWORD: data[CONF_PASSWORD],
-            CONF_URL: data.get(CONF_URL, EU_URL),
-            CONF_TYPE: ATTR_TYPE_CLOUD,
-        }
-
-        return self.async_create_entry(title=data[CONF_USERNAME], data=auth_data)
-
-    async def _validate_and_create_camera_rtsp(self, data):
+    async def _validate_and_create_camera_rtsp(self, data: dict) -> FlowResult:
         """Try DESCRIBE on RTSP camera with credentials."""
 
         # Get EZVIZ cloud credentials from config entry
-        ezviz_client_creds = {
-            CONF_USERNAME: None,
-            CONF_PASSWORD: None,
-            CONF_URL: None,
+        ezviz_token = {
+            CONF_SESSION_ID: None,
+            CONF_RFSESSION_ID: None,
+            "api_url": None,
         }
+        ezviz_timeout = DEFAULT_TIMEOUT
 
         for item in self._async_current_entries():
             if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
-                ezviz_client_creds = {
-                    CONF_USERNAME: item.data.get(CONF_USERNAME),
-                    CONF_PASSWORD: item.data.get(CONF_PASSWORD),
-                    CONF_URL: item.data.get(CONF_URL),
+                ezviz_token = {
+                    CONF_SESSION_ID: item.data.get(CONF_SESSION_ID),
+                    CONF_RFSESSION_ID: item.data.get(CONF_RFSESSION_ID),
+                    "api_url": item.data.get(CONF_URL),
                 }
+                ezviz_timeout = item.data.get(CONF_TIMEOUT, DEFAULT_TIMEOUT)
 
         # Abort flow if user removed cloud account before adding camera.
-        if ezviz_client_creds[CONF_USERNAME] is None:
+        if ezviz_token.get(CONF_SESSION_ID) is None:
             return self.async_abort(reason="ezviz_cloud_account_missing")
 
+        ezviz_client = EzvizClient(token=ezviz_token, timeout=ezviz_timeout)
+
         # We need to wake hibernating cameras.
         # First create EZVIZ API instance.
-        try:
-            ezviz_client = await self.hass.async_add_executor_job(
-                _get_ezviz_client_instance, ezviz_client_creds
-            )
-
-        except InvalidURL as err:
-            raise InvalidURL from err
-
-        except HTTPError as err:
-            raise InvalidHost from err
+        await self.hass.async_add_executor_job(ezviz_client.login)
 
-        except PyEzvizError as err:
-            raise PyEzvizError from err
-
-        # Secondly try to wake hibernating camera.
-        try:
-            await self.hass.async_add_executor_job(
-                ezviz_client.get_detection_sensibility, data[ATTR_SERIAL]
-            )
-
-        except HTTPError as err:
-            raise InvalidHost from err
+        # Secondly try to wake hybernating camera.
+        await self.hass.async_add_executor_job(
```

Comment:
Combine sequential calls into the executor into one call. Make a function that does all the sync calls and schedule that function on the executor. It's expensive to switch context and use the executor.

Reply:
Addressed in https://github.com/home-assistant/core/pull/94384